File: 950.py
from collections import deque
from functools import cache

class Solution:
    @staticmethod
    @cache
    def _dd_(n):
        q = deque(range(n))
        st = []
        while len(q) != 0:
            st.append(q.popleft())
            q.rotate(-1)
        return tuple(st)

    def deckRevealedIncreasing(self, deck: List[int]) -> List[int]:
        deck.sort()
        res = [None] * len(deck)
        for i, x in enumerate(self._dd_(len(deck))):
            res[x] = deck[i]
        return res

File: sol/950.cpp
#include <queue>

struct Solution {
    vector<int> deckRevealedIncreasing(vector<int>& deck) {
        sort(deck.begin(), deck.end());
        vector<int> res(deck.size());
        queue<int> q;
        for (int i = 0; i < deck.size(); ++i){
            q.push(i);
        }
        int idx = 0;
        while (!q.empty()){
            res[q.front()] = deck[idx++];
            q.pop();
            q.push(q.front());
            q.pop();
        }
        return res;
    }
};
